syntax = "proto3";

package connect;


service UserService {
  rpc GetUser(UserRequest) returns (UserResponse);
  rpc CreateUser(UserRequest) returns (UserResponse);
  rpc ListUsers(Empty) returns (stream UserResponse);
}

message User {
  string id = 1;
  string uid = 2;
  string uuid = 3;
  string username = 4;
  string display_name = 5;
  string profile_image_url = 6;
  string phone_number = 7;
  string bio = 8;

  bool is_online = 9;
  bool has_story = 10;
  int64 last_seen = 11;
  int64 created_at = 12;
  int64 last_login = 13;

  repeated string group_id = 14;
  repeated string past_uuids = 15;
  repeated string blocked_users_id = 16;

  bool is_verified = 17;
  bool is_active = 18;
  bool is_staff = 19;
}

message UserRequest {
  string uid = 1;
}

message UserResponse {
  User user = 1;
}

message Empty {}


